pkgs/top-level/all-packages.nix

```diff
@@ -1004,6 +1004,36 @@ with pkgs;
     ];
   } ../build-support/setup-hooks/validate-pkg-config.sh;
 
+  autoPatchPcHook =
+    makeSetupHook
+      {
+        name = "auto-patch-pc-hook";
+        propagatedBuildInputs = [ pkg-config ];
+      }
+      (
+        writeScript "patch-pc.sh" ''
+          fixupOutputHooks+=(autoPatchPcHook)
+
+          autoPatchPcHook() (
+              # Some packages list their own libraries as deps, they should be searched first in case of cyclic deps
+              export PKG_CONFIG_PATH_${pkgconf.suffixSalt}="$prefix/lib/pkgconfig:$prefix/share/pkgconfig:$PKG_CONFIG_PATH"
+
+              shopt -s nullglob
+              for pc in $prefix/{lib,lib64,share}/pkgconfig/*.pc; do
+                  echo Fixing require paths in $pc file
+
+                  if ! ${lib.getExe python3Minimal} \
+                    ${../build-support/setup-hooks/patch-pc.py} \
+                    ${lib.getExe pkgconf} "$pc" > "$pc.patched"; then
```

Comment:
Is this actually intended to be a different `pkgconf` than above propagated `pkg-config`? Might it make more sense to make a `let...in` for the whole block and use the same here as in propagation?